### DISCUSSION TITLE
where 'use' hook type intergrated in next.js  and where 'use' hook is implemented?

### DISCUSSION THREAD

**Discussion by dante01yoon** [vercel/next.js#51473] [App Router]
Why `use` hook is only works in next.js13 app directory and not normal CSR react app?

![스크린샷 2023-06-18 오후 8 39 29](https://github.com/vercel/next.js/assets/6510430/b51a8c96-6881-4659-8d92-63ec1a31553c)

where above next.d.ts is extended with react module in nextjs? 

## nextjs  type check throws no error 
// "@types/react": "^18.0.22",
//  "react": "^18.2.0",
![스크린샷 2023-06-18 오후 8 39 53](https://github.com/vercel/next.js/assets/6510430/cd2d11ac-0149-47f5-8087-812341e5faa8)

## react CRA app throws type error
//  "react": "^18.2.0",
//  "@types/react": "^18.0.28",

![스크린샷 2023-06-18 오후 8 39 59](https://github.com/vercel/next.js/assets/6510430/8dd3c21e-8d31-4a18-a386-786fac4d8b17)

**Comment by icyJoseph** [comment]
- next.d.ts https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react
- use hook https://github.com/facebook/react/blob/main/packages/react/src/ReactHooks.js#L225

However, how it is manifested as an actual function I am not sure. Maybe? https://nextjs.org/blog/next-13-4#what-does-app-router-stability-mean:

> The App Router is built on top of the React canary channel, which is now ready for framework adoption of features like Server Components. [Learn more](https://react.dev/blog/2023/05/03/react-canaries).

And, https://react.dev/blog/2023/05/03/react-canaries, but I'd be guessing. Not sure where in the app route code architecture one would find this, but I think you get an idea.

**Reply by dante01yoon** [reply]
Thank you for your comments! I'll take a look
